lib/smarter_csv/smarter_csv.rb

```diff
@@ -186,6 +186,35 @@ def SmarterCSV.process(input, options={}, &block)   # first parameter: filename
     end
   end
 
+  # SmarterCSV.write(rows, out, header = nil)
+  #   `rows`   -> Array of hashes representing each row.
+  #   `out`    -> The filename to write the data to.
+  #   `header` -> Array of column names as strings or symbols. Optional. If it
+  #               isn't given, the keys of the first row are used as header.
+  #
+  # Example:
+  #   rows = [{"a" => 1, "b" => 2}, {"a" => 4, "b" => 5, "c" => 99}]
+  #   out = "results.csv"
+  #   SmarterCSV.write(rows, out)
+  #
+  #   $ cat results.csv
+  #   a,b
+  #   1,2
+  #   4,5
+  def SmarterCSV.write(rows, out, header = nil)
+    unless header
+      header = rows.first.keys
```

Comment:
two issues:
* this assumes that the first record has all keys... depending on where that data came from, this might not  be the case
* this assumes that the order of the key/value pairs in all the hashes (rows) is the same

Probably better to define the headers separatetly as an array, then emit the row depending on that sort-order and headers